src/compas_fab/backends/pybullet/backend_features/pybullet_forward_kinematics.py

```diff
@@ -11,39 +11,38 @@ def __init__(self, client):
-        # """Calculate the robot's forward kinematic.
-        #
-        # Parameters
-        # ----------
-        # configuration : :class:`compas_fab.robots.Configuration`
-        #     The full configuration to calculate the forward kinematic for. If no
-        #     full configuration is passed, the zero-joint state for the other
-        #     configurable joints is assumed.
-        # group : str, optional
-        #     Unused parameter.
-        # options : dict, optional
-        #     Dictionary containing the following key-value pairs:
-        #
-        #     - ``"base_link"``: (:obj:`str`) The name of the base link.
-        #     - ``"ee_link"``: (:obj:`str`, optional) The name of the link to
-        #       calculate the forward kinematics for. Defaults to the group's end
-        #       effector link.
-        #
-        # Returns
-        # -------
-        # :class:`Frame`
-        #     The frame in the world's coordinate system (WCF).
-        # """
-        raise NotImplementedError
-        # """
-        # """
-        # ee_link_name = options['ee_link_name']
-        # joints = self.client.joints_from_names(self.client.robot_uid, configuration.joint_names)
-        # ee_link = self.client.link_from_name(self.client.robot_uid, ee_link_name)
-        # self.client.set_joint_positions(self.client.robot_uid, joints, configuration.values)
-        # pose = self.client.get_link_pose(self.client.robot_uid, ee_link)
-        # if options.get('check_collision'):
-        #     collision, names = self.client.client.collision_check()
-        #     if collision:
-        #         raise CollisionError(*names)
-        # return frame_from_pose(pose)
+        """Calculate the robot's forward kinematic.
+
+        Parameters
+        ----------
+        configuration : :class:`compas_fab.robots.Configuration`
+            The full configuration to calculate the forward kinematic for. If no
+            full configuration is passed, the zero-joint state for the other
+            configurable joints is assumed.
+        group : str, optional
+            The planning group used for calculation. Defaults to the robot's
+            main planning group.
+        options : dict, optional
+            Dictionary containing the following key-value pairs:
+
+            - ``"robot"``: (:class:`compas_fab.robots.Robot`) The robot whose
+              kinematics are to be calculated.
+            - ``"link"``: (:obj:`str`, optional) The name of the link to
+              calculate the forward kinematics for. Defaults to the end effector.
+            - ``"check_collision"``: (:obj:`str`, optional) When ``True``,
+              :meth:`compas_fab.backends.PyBulletClient.check_collisions` will be called.
+              Defaults to ``False``.
+
+        Returns
+        -------
+        :class:`Frame`
+            The frame in the world's coordinate system (WCF).
+        """
+        robot = options['robot']
```

Comment:
Joint names, but not link names, are recoverable from the pybullet server.  My intention is that a user doesn't have to care about pybullet's internal id system, except for maybe the robot's uid.  I would put just the robot uid in the options (rather than the full robot)  if it were possible to recover the links, but it's not.